src/librustc_mir/dataflow/move_paths/builder.rs

```diff
@@ -380,7 +380,9 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
                 self.gather_operand(discr);
             }
 
-            TerminatorKind::Yield { ref value, .. } => {
+            TerminatorKind::Yield { ref value, resume_arg: ref place, .. } => {
+                self.create_move_path(place);
+                self.gather_init(place.as_ref(), InitKind::Deep);
```

Comment:
I noticed that the order of `gather_init` and `gather_operand` is swapped compared to `DropAndReplace`, but I'm not sure if it matters.